parachains/runtimes/assets/common/src/matching.rs

```diff
@@ -0,0 +1,90 @@
+// Copyright (C) 2023 Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use cumulus_primitives_core::ParaId;
+use frame_support::{
+	pallet_prelude::Get,
+	traits::{Contains, ContainsPair},
+};
+use xcm::{
+	latest::prelude::{MultiAsset, MultiLocation},
+	prelude::*,
+};
+
+pub struct StartsWith<T>(sp_std::marker::PhantomData<T>);
+impl<Location: Get<MultiLocation>> Contains<MultiLocation> for StartsWith<Location> {
+	fn contains(t: &MultiLocation) -> bool {
+		t.starts_with(&Location::get())
+	}
+}
+
+pub struct Equals<T>(sp_std::marker::PhantomData<T>);
+impl<Location: Get<MultiLocation>> Contains<MultiLocation> for Equals<Location> {
+	fn contains(t: &MultiLocation) -> bool {
+		t == &Location::get()
+	}
+}
+
+pub struct StartsWithExplicitGlobalConsensus<T>(sp_std::marker::PhantomData<T>);
+impl<Network: Get<NetworkId>> Contains<MultiLocation>
+	for StartsWithExplicitGlobalConsensus<Network>
+{
+	fn contains(t: &MultiLocation) -> bool {
+		match t.interior.global_consensus() {
+			Ok(requested_network) if requested_network.eq(&Network::get()) => true,
+			_ => false,
+		}
+	}
+}
+
+frame_support::parameter_types! {
+	pub LocalMultiLocationPattern: MultiLocation = MultiLocation::new(0, Here);
+	pub ParentLocation: MultiLocation = MultiLocation::parent();
+}
+
+/// Accepts an asset if it is from the origin.
+pub struct IsForeignConcreteAsset<IsForeign>(sp_std::marker::PhantomData<IsForeign>);
+impl<IsForeign: ContainsPair<MultiLocation, MultiLocation>> ContainsPair<MultiAsset, MultiLocation>
+	for IsForeignConcreteAsset<IsForeign>
+{
+	fn contains(asset: &MultiAsset, origin: &MultiLocation) -> bool {
+		log::trace!(target: "xcm::contains", "IsForeignConcreteAsset asset: {:?}, origin: {:?}", asset, origin);
+		matches!(asset.id, Concrete(ref id) if IsForeign::contains(id, &origin))
+	}
+}
+
+/// Checks if 'a' is from sibling location `b`, so means that `MultiLocation-a' starts with `MultiLocation-b'
```

Comment:
These docs are not correct :P 

I mean this is one part and then it also checks that it comes from a parachain that is a sibling.